File: bindings/java/src/main/java/com/psyne/CompressionType.java
package com.psyne;

/**
 * Compression algorithms supported by Psyne.
 * 
 * This enum defines the different compression algorithms that can be used
 * to compress messages in Psyne channels.
 */
public enum CompressionType {
    /**
     * No compression.
     * Messages are sent as-is without any compression.
     */
    NONE(0),
    
    /**
     * LZ4 compression.
     * Fast compression algorithm with reasonable compression ratios.
     */
    LZ4(1),
    
    /**
     * Zstandard compression.
     * Modern compression algorithm with excellent compression ratios.
     */
    ZSTD(2),
    
    /**
     * Snappy compression.
     * Very fast compression algorithm optimized for speed.
     */
    SNAPPY(3);
    
    private final int value;
    
    CompressionType(int value) {
        this.value = value;
    }
    
    /**
     * Gets the native value of this compression type.
     * 
     * @return The native value
     */
    public int getValue() {
        return value;
    }
}

File: bindings/csharp/src/Psyne/CompressionType.cs
namespace Psyne
{
    /// <summary>
    /// Specifies the compression algorithm to use for messages.
    /// </summary>
    public enum CompressionType
    {
        /// <summary>
        /// No compression applied.
        /// </summary>
        None = 0,

        /// <summary>
        /// LZ4 compression algorithm.
        /// Fast compression and decompression with good compression ratios.
        /// </summary>
        LZ4 = 1,

        /// <summary>
        /// Zstandard (Zstd) compression algorithm.
        /// Excellent compression ratios with good performance.
        /// </summary>
        Zstd = 2,

        /// <summary>
        /// Snappy compression algorithm.
        /// Very fast compression with moderate compression ratios.
        /// </summary>
        Snappy = 3
    }
}